src/Magnum/Math/Random.h

```diff
@@ -0,0 +1,131 @@
+#include "Magnum/Math/Constants.h"
+#include "Magnum/Math/Vector2.h"
+#include "Magnum/Math/Vector3.h"
+#include "Magnum/Math/Quaternion.h"
+#include "Magnum/Math/Functions.h"
+
+namespace Magnum
+{
+namespace Math
+{
+
+namespace Implementation
+{
+static std::seed_seq seeds{{
+    static_cast<std::uintmax_t>(std::random_device{}()),
+    static_cast<std::uintmax_t>(std::chrono::steady_clock::now()
+                                    .time_since_epoch()
+                                    .count()),
+}};
+
+class RandomGenerator
+{
+public:
+    RandomGenerator() = delete;
+
+    template <typename T>
+    static typename std::enable_if<std::is_same<Int, T>::value, T>::type
+    generate(T start = -Magnum::Math::Constants<T>::inf(),
+             T end = Magnum::Math::Constants<T>::inf())
+    {
+        return std::uniform_int_distribution<T>{start, end}(generator());
+    }
+
+    template <typename T>
+    static typename std::enable_if<std::is_same<Float, T>::value, T>::type
+    generate(T start = -Magnum::Math::Constants<T>::inf(),
+             T end = Magnum::Math::Constants<T>::inf())
+    {
+        return std::uniform_real_distribution<T>{start, end}(generator());
+    }
+
+public:
+    static std::mt19937 &generator()
+    {
+        static std::mt19937 g{seeds};
+        return g;
+    }
+};
+} // namespace Implementation
+namespace Random
+{
+template <class T = Float>
+T randomUnsignedScalar() // range [0, 1]
+{
+    return Implementation::RandomGenerator::generate<T>(static_cast<T>(0.0f),
+                                                        static_cast<T>(1.0f));
+}
+template <class T = Float>
+T randomSignedScalar() // range [-1, 1]
+{
+    return Implementation::RandomGenerator::generate(static_cast<T>(-1.0f), static_cast<T>(1.0f));
+}
+
+template <class T = Float>
+Vector2<T> randomUnitVector2()
+{
+    auto a = Implementation::RandomGenerator::generate(0.0f, 2 * Math::Constants<T>::pi());
+    return {std::cos(a), std::sin(a)};
+}
+
+template <class T = Float>
+Vector3<T> randomUnitVector3()
+{
+    auto a = Implementation::RandomGenerator::generate(0.0f, 2 * Math::Constants<T>::pi());
+    auto z = randomSignedScalar();
+    auto r = sqrt<T>(1 - z * z);
+    return {r * std::cos(a), r * std::sin(a), z};
+}
+template <class T = Float>
+Vector2<T> randomPointInACircle() // always length < 1
+{
+    while (true)
+    {
+        auto p = Vector2<T>(
+            randomSignedScalar(), randomSignedScalar());
+        if (p.length() >= 1)
+            continue;
+        return p;
+    }
+}
+
+template <class T = Float>
+Vector3<T> randomPointInASphere() // always length < 1
+{
+
+    while (true)
+    {
+        auto p = Vector3<T>(randomSignedScalar(),
+                            randomSignedScalar(),
+                            randomSignedScalar());
+        if (p.length() >= 1)
+            continue;
+        return p;
+    }
+}
+
+bool randomBool()
+{
+    return static_cast<bool>(randomUnsignedScalar<Int>());
+}
```

Comment:
Hmm, do we actually need this? :) The `randomUnsignedScalar()` for integers isn't really useful either. Maybe instead add a `static_assert(IsFloatingPoint<T>::value, "");` to the RandomGenerator and remove everything related to integers? Those can be always added back later. (The trait is in [Magnum/Math/TypeTraits.h](https://doc.magnum.graphics/magnum/structMagnum_1_1Math_1_1IsFloatingPoint.html).)

Reply:
Well, like i said in the beginning, a initiation for a new magnum functionality :) Can grow later with demand :) 

Deleting/Fixing. :)

Reply:
I first deleted, and then put it back haha :D 
From our initial discussion:
```
Float randomUnsignedScalar(); // range [0, 1]
Float randomSignedScalar(); // range [-1, 1]
Vector2 randomPointOnACircle(); // always length = 1
Vector3 randomPointOnASphere(); // also
Quaternion randomRotation(); // always normalized
```

Maybe keeping it with integers to not to cast anything if we roll a dice?(`randomUnsignedScalar<int>(1,6)`)or to see if that girl is single?(`randomUnsignedScalar<int>(0,1)`[bool to cast, yes. maybe type traits for numerics?])